modules/libcom/src/iocsh/iocsh.cpp

```diff
@@ -1474,6 +1476,43 @@ static void iocshLoadCallFunc(const iocshArgBuf *args)
     iocshSetError(iocshLoad(args[0].sval, args[1].sval));
 }
 
+/* set*/
+static const iocshArg iocshSetArg0 = { "name",iocshArgString};
+static const iocshArg iocshSetArg1 = { "value", iocshArgString};
+static const iocshArg *iocshSetArgs[2] = {&iocshSetArg0, &iocshSetArg1};
+static const iocshFuncDef iocshSetFuncDef = {"set",2,iocshSetArgs,
+                                              "Takes name and value\n"
+                                              "Sets ioc shell variable to value\n"};
+static void iocshSetCallFunc(const iocshArgBuf *args)
+{
+    iocshContext *context;
+
+    if (iocshContextId) {
+        context = (iocshContext *) epicsThreadPrivateGet(iocshContextId);
+
+        if (context != NULL) {
+            macPutValue(context->handle, args[0].sval, args[1].sval);
```

Comment:
I discovered I can segfault the IOC by running the command `set` with no parameters, please add a check for `args[0].sval == NULL`. However it occurred to me that we could omit the `show` command completely, and just have `set` with no parameters display the list of variables, like we already do with `var`.

However, the following commands _should_ be allowed as the second `set` deletes the variable that was created by the first one.
```
set abcd 1234
set abcd
```

Reply:
I lied about the segfault above, I just found my notes and it required these:
```
set abcd 1234
set
```
The reason is still the same, `macPutValue()` doesn't like the parameter name to be NULL.

Reply:
@anjohnson I am working on it, and will test. Will write here if have questions.